src/cbmc_starter_kit/update.py

```diff
@@ -0,0 +1,184 @@
+                              f'CBMC root {args.cbmc_root}')
+    else:
+        args.no_migrate = True
+    logging.debug('CBMC starter kit root: %s', args.starter_kit_root)
+    return args
+
+################################################################
+
+def files_under_root(root=None, symlinks_only=False):
+    cmd = ['find', '.']
+    if symlinks_only:
+        cmd += ['-type', 'l']
+    kwds = {
+        'cwd': root,
+        'text': True,
+        'capture_output': True,
+    }
+    files = subprocess.run(cmd, **kwds, check=True).stdout.splitlines()
+    return sorted(files)
+
+################################################################
+
+def migrate(cbmc_root, starter_kit_root):
+    logging.debug('Migrating CBMC starter kit')
+    templates = files_under_root(starter_kit_root/util.REPOSITORY_TEMPLATES)
+    logging.debug('Migrating CBMC starter kit: found templates: %s', templates)
+    symlinks = files_under_root(cbmc_root, symlinks_only=True)
+    logging.debug('Migrating CBMC starter kit: found symlinks: %s', symlinks)
+
+    cbmc_symlinks = [path for path in templates if path in symlinks]
+    for symlink in cbmc_symlinks:
+        cbmc_link = cbmc_root / symlink
+        cbmc_file = cbmc_link.resolve()
+        logging.warning('Copying: %s -> %s', cbmc_file, cbmc_link)
+        assert cbmc_file.exists()
+        assert cbmc_link.exists()
+        cbmc_link.unlink()
+        shutil.copy(cbmc_file, cbmc_link)
+
+def remove_negative_tests(cbmc_root):
+    logging.debug('Removing CBMC starter kit negative tests')
+    negative_tests = cbmc_root / 'negative_tests'
+    if negative_tests.is_dir():
+        logging.warning('Removing: %s', negative_tests)
+        shutil.rmtree(negative_tests)
+
+def update(cbmc_root):
+    logging.debug('Updating CBMC starter kit')
+    for path in [f'{util.PROOF_DIR}/{util.COMMON_MAKEFILE}', f'{util.PROOF_DIR}/{util.RUN_SCRIPT}']:
+        src = util.package_repository_template_root() / path
+        dst = cbmc_root / path
+        logging.warning('Copying: %s -> %s', src, dst)
+        version.copy_with_version(src, dst)
+
+def remove_starter_kit_submodule(cbmc_root, force=False):
+    logging.debug('Checking for starter kit submodule')
+    starter_kit = repository.starter_kit_root(repo=cbmc_root, abspath=False)
+    if not starter_kit:
+        logging.debug('Starter kit submodule does not exist')
+        return
+    if not force:
+        logging.warning('Use --force if you want to remove the starter kit submodule: %s',
+                        starter_kit)
+        return
+    for submodule in git.Repo(cbmc_root, search_parent_directories=True).submodules:
+        if submodule.path == str(starter_kit):
+            logging.warning('Removing: starter kit submodule: %s', starter_kit)
+            submodule.remove()
+            return
+    logging.error('This should be impossible: Failed to remove starter kit submodule: %s',
```

Comment:
Add `sys.exit(1)` ?

Reply:
I'm not sure.  This should be impossible, but if the submodule is not removed, the user can still trust the results to run the proofs correctly.